x-pack/plugins/fleet/server/services/epm/packages/get.ts

```diff
@@ -169,6 +170,45 @@ export const getPackageUsageStats = async ({
   };
 };
 
+export async function getPackagePermissions(
+  soClient: SavedObjectsClientContract,
+  pkgName: string,
+  pkgVersion: string,
+  namespace = '*'
+): Promise<PackagePermissions | undefined> {
+  const pkg = await getPackageInfo({ savedObjectsClient: soClient, pkgName, pkgVersion });
+  if (!pkg.data_streams) {
+    return undefined;
+  }
+
+  const clusterPermissions = new Set<string>();
+  const indices: PackagePermissions['indices'] = pkg.data_streams!.map((ds) => {
+    if (ds.permissions?.cluster) {
+      ds.permissions.cluster.forEach((p) => clusterPermissions.add(p));
+    } else {
+      clusterPermissions.add('monitor');
+    }
+
+    let index = `${ds.type}-${ds.dataset}-${namespace}`;
+    if (ds.dataset_is_prefix) {
```

Comment:
APM uses `data_is_prefix`. I'm not sure if the way I'm building the pattern is correct though. I'm assuming the prefix includes the namespace. So for a namespace `default`, the prefix should be something like `logs-apm.thing-default-*`